### PR TITLE
[DDW-349] Disabled ITN redemption before being synced

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR disabled ITN redemption in case Daedalus is not fully synced.

## Screenshots

Screenshot 2020-08-11 at 13 28 58

Screenshot 2020-08-11 at 13 29 17

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/containers/staking/dialogs/redeem-itn-rewards/Step1ConfigurationContainer.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ export default class Step1ConfigurationContainer extends Component<Props> {
 
   render() {
     const { onClose, onBack, stores, actions } = this.props;
+    const { isSynced, syncPercentage } = stores.networkStatus;
```
Comment (daniloprates) [inline source/renderer/app/containers/staking/dialogs/redeem-itn-rewards/Step1ConfigurationContainer.js:20]: It makes more sense to have this logic inside **RedeemItnRewardsContainer.js**, as it already has the **NoWalletsContainer.js**

File: source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.js [hunk 2]
```diff
@@ -313,58 +338,73 @@ export default class Step1ConfigurationDialog extends Component<Props> {
...
-            <WalletsDropdown
-              className={styles.walletsDropdown}
-              {...walletsDropdownField.bind()}
-              numberOfStakePools={4}
-              wallets={wallets}
-              onChange={onSelectWallet}
-              placeholder={intl.formatMessage(
-                messages.selectWalletInputPlaceholder
+          {!isSynced ? (
+            <p className={styles.redemptionUnavailable}>
```
Comment (daniloprates) [inline source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.js:342]: I'd move this into a dedicated component, like the **NoWalletsDialog.js**, as you're adding a lot of information that is actually not related to the Step 1 - Configuration.

File: source/renderer/app/containers/staking/dialogs/redeem-itn-rewards/Step1ConfigurationContainer.js (hunk 1 above)
Comment (nikolaglumac) [reply source/renderer/app/containers/staking/dialogs/redeem-itn-rewards/Step1ConfigurationContainer.js:20]: @daniloprates I do not agree. If we do that then it could happen that sync progress drops below 100% in the middle of the redemption process in which case we would revert to that "Not in sync" step and that would be wrong.

File: source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.js (hunk 2 above)
Comment (nikolaglumac) [reply source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.js:342]: @daniloprates please see https://github.com/input-output-hk/daedalus/pull/2146#discussion_r468632031
